### PR TITLE
Issue 4712 - ticket groups bugs

### LINKED ISSUE #4712: Ticket Groups: bugs
### Bugs
1. When all the meshes are hidden, save to current view creates a manual coloured group with no objects
2. On a new ticket, calling "save to current view" creates some groups cannot be edited. Editing them creates a different group instead

### Description
- These are happening in: both
- These require a hotfix to production: no

### Steps to replicate
- In the viewer, open a new ticket;
- Hide all the meshes;
- Save to current view;
> Bug 1: a colored **manual** group is created with no objects

- Try to edit either the hidden or the colored group
> Bug 2: instead of editing the group, a new one is created

### Pictures/ Gifs of the error

https://github.com/3drepo/3drepo.io/assets/43547703/e59c69b0-0a87-4cd3-95e5-4c7d7ac35fb3

### PR DESCRIPTION
**Pull request by Amantini1997**
This fixes #4712

#### Description
- The highlighted objects can be hidden. If that's the case, the equivalent group is not created on save/update current view
- Groups via save/update current view in a new ticket were missing the key

#### Test cases
Click on "new ticket" and select a template with view and:
1. Highlight a mesh
2. Hide that mesh
3. Save the current view
> No error should be triggered
4. Open the groups (there should be a hidden group)
5. Edit the hidden group
> The group should be edited correctly and no other groups should be created

### CODE REVIEW

File: frontend/src/v5/helpers/viewpoint.helpers.ts
Comment (The-Daniel) [inline frontend/src/v5/helpers/viewpoint.helpers.ts:1]: I think this might be the same bug as #4710
Sometimes when clicking 'edit' on a hidden group it is saying there are 0 objects when you can see in the viewer there are some highlighted objects. Sometimes the number is accurate. I'm not entirely sure how to reproduce every time
image

Comment (The-Daniel) [inline frontend/src/v5/helpers/viewpoint.helpers.ts:1]: Sometimes just making the form dirty validates it.
image
image

**Review by The-Daniel — CHANGES_REQUESTED** [review]
I know a lot of these are unrelated to the original issue, but since they fit in to the vague PR title I thought I'd post here. You might want to create new issues for them though

File: frontend/src/v5/helpers/viewpoint.helpers.ts
[the remark this answers is not in the document]
Comment (Amantini1997) [reply frontend/src/v5/helpers/viewpoint.helpers.ts:1]: I noticed it this morning, but I was only able to reproduce it on a particular model, so I thought the problem was in the model. Btw, as I was debugging it, I added a breakpoint right before the line making the API call (and so slowing down the flow in the frontend), the problem was not happening anymore. I suspect it could be a backend bug, but I am not sure

Comment (The-Daniel) [inline frontend/src/v5/helpers/viewpoint.helpers.ts:1]: I have a model with 8000 objects. When editing a group with many selected objects the object count doesn't get higher than 5000 even though the correct number appear highlighted in the viewer.
image

File: frontend/src/v5/helpers/viewpoint.helpers.ts [hunk 1]
```diff
@@ -86,7 +86,10 @@ export const getViewerState = async () => {
 	}
 
 	if (viewpointV4.highlighted_group) {
-		state.colored = (state.colored || []).concat(convertToV5GroupOverride(viewpointV4.highlighted_group, ViewpointGroupOverrideType.COLORED));
+		state.colored = (state.colored || [])
+			.concat(convertToV5GroupOverride(viewpointV4.highlighted_group, ViewpointGroupOverrideType.COLORED))
+			// eslint-disable-next-line no-underscore-dangle
+			.filter(({ group }) => !!(group as Group).objects.filter((obj) => obj._ids.length).length);
```
Comment (sanmont3drepo) [inline frontend/src/v5/helpers/viewpoint.helpers.ts:92]: Why did you change this?

Comment (Amantini1997) [reply frontend/src/v5/helpers/viewpoint.helpers.ts:92]: To fix this bug
image

**Review by sanmont3drepo — APPROVED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Review by The-Daniel — APPROVED** [review]
